spec/lib/two_factor_authentication/models/two_factor_authenticatable_spec.rb

```diff
@@ -44,33 +44,76 @@
       end
     end
 
-    it_behaves_like 'otp_code', GuestUser.new
-    it_behaves_like 'otp_code', EncryptedUser.new
+    it_behaves_like 'generate_totp_code', GuestUser.new
+    it_behaves_like 'generate_totp_code', EncryptedUser.new
   end
 
-  describe '#authenticate_otp' do
-    shared_examples 'authenticate_otp' do |instance|
+  describe '#create_direct_otp' do
+    let(:instance) { build_guest_user }
+
+    it 'creates an otp_code' do
+      expect(instance.otp_code).to be_nil
+      instance.create_direct_otp
+      expect(instance.otp_code).not_to be_nil
+    end
```

Comment:
Since the `create_direct_otp` method updates the `direct_otp` column in the DB, I would expect this test to make sure the `direct_otp` column was updated, as opposed to indirectly checking it via `otp_code`.